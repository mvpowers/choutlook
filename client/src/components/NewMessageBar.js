import React from 'react';
import styled from 'styled-components';
import { FaBars, FaPlus } from 'react-icons/lib/fa';

const Container = styled.div`
  display: flex;
  flex-direction: row;
  background-color: #f4f4f4;
  color: #5084aa;
`;

const Icon = styled.div`
  padding: 15px;

  &:hover {
    background-color: #c7e0f4;
  }
`;

const NewMessage = styled.div`
  display: flex;
  align-items: center;
  padding-right: 15px;

  &:hover {
    background-color: #c7e0f4;
  }
`;

export default () => (
  <Container>
    <Icon>
      <FaBars size={16} />
    </Icon>
    <NewMessage>
      <Icon>
        <FaPlus size={16} />
      </Icon>
      <div>New message</div>
    </NewMessage>
  </Container>
);
