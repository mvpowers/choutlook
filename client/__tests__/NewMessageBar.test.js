import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NewMessageBar from '../src/components/NewMessageBar';

Enzyme.configure({ adapter: new Adapter() });

describe('NewMessageBar', () => {
  it('should be defined', () => {
    expect(NewMessageBar).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(<NewMessageBar />);
    expect(tree).toMatchSnapshot();
  });
});
