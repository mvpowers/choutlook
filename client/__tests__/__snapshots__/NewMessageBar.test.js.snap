// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMessageBar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <NewMessageBar__Icon>
          <FaBars
            size={16}
          />
        </NewMessageBar__Icon>,
        <NewMessageBar__NewMessage>
          <NewMessageBar__Icon>
            <FaPlus
              size={16}
            />
          </NewMessageBar__Icon>
          <div>
            New message
          </div>
        </NewMessageBar__NewMessage>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FaBars
            size={16}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": 16,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NewMessageBar__Icon>
              <FaPlus
                size={16}
              />
            </NewMessageBar__Icon>,
            <div>
              New message
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <FaPlus
                size={16}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "size": 16,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New message",
            },
            "ref": null,
            "rendered": "New message",
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <NewMessageBar__Icon>
            <FaBars
              size={16}
            />
          </NewMessageBar__Icon>,
          <NewMessageBar__NewMessage>
            <NewMessageBar__Icon>
              <FaPlus
                size={16}
              />
            </NewMessageBar__Icon>
            <div>
              New message
            </div>
          </NewMessageBar__NewMessage>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FaBars
              size={16}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 16,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NewMessageBar__Icon>
                <FaPlus
                  size={16}
                />
              </NewMessageBar__Icon>,
              <div>
                New message
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <FaPlus
                  size={16}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "size": 16,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New message",
              },
              "ref": null,
              "rendered": "New message",
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
